src/java/org/apache/poi/hssf/record/FilePassRecord.java

```
@@ -88,32 +88,36 @@ public FilePassRecord(RecordInputStream in) {
-                break;
-            case cryptoAPI:
-                out.writeShort(encryptionInfo.getVersionMajor());
-                out.writeShort(encryptionInfo.getVersionMinor());
-                out.writeInt(encryptionInfo.getEncryptionFlags());
-                ((CryptoAPIEncryptionHeader)encryptionInfo.getHeader()).write(bos);
-                ((CryptoAPIEncryptionVerifier)encryptionInfo.getVerifier()).write(bos);
-                break;
-            default:
-                throw new EncryptedDocumentException("not supported");
-        }
+            switch (encryptionInfo.getEncryptionMode()) {
+                case xor:
+                    ((XOREncryptionHeader)encryptionInfo.getHeader()).write(bos);
+                    ((XOREncryptionVerifier)encryptionInfo.getVerifier()).write(bos);
+                    break;
+                case binaryRC4:
+                    out.writeShort(encryptionInfo.getVersionMajor());
+                    out.writeShort(encryptionInfo.getVersionMinor());
+                    ((BinaryRC4EncryptionHeader)encryptionInfo.getHeader()).write(bos);
+                    ((BinaryRC4EncryptionVerifier)encryptionInfo.getVerifier()).write(bos);
+                    break;
+                case cryptoAPI:
+                    out.writeShort(encryptionInfo.getVersionMajor());
+                    out.writeShort(encryptionInfo.getVersionMinor());
+                    out.writeInt(encryptionInfo.getEncryptionFlags());
+                    ((CryptoAPIEncryptionHeader)encryptionInfo.getHeader()).write(bos);
+                    ((CryptoAPIEncryptionVerifier)encryptionInfo.getVerifier()).write(bos);
+                    break;
+                default:
+                    throw new EncryptedDocumentException("not supported");
+            }
 
-        out.write(data, 0, bos.getWriteIndex());
+            out.write(data, 0, bos.getWriteIndex());
+        } catch (IOException ioe) {
+            // should never happen in practice
```

doing such a catch looks could swallow exceptions in the future, I'd rather wrap the checked exception in an unchecked one if we know this is not expected anytime, e.g. IllegalStateException, so we still see that it happens if it should still happen at some point later